apps/dashboard/test/apps/nav_bar_test.rb

```diff
@@ -0,0 +1,142 @@
+require 'test_helper'
+
+class NavBarTest < ActiveSupport::TestCase
+
+  def setup
+    # Mock the sys installed applications
+    OodAppkit.stubs(:clusters).returns(OodCore::Clusters.load_file('test/fixtures/config/clusters.d'))
+    SysRouter.stubs(:base_path).returns(Rails.root.join("test/fixtures/sys_with_gateway_apps"))
+  end
+
+  test "NavBar.items should return navigation template when nav_item is a string matching a static link" do
+    nav_item = "all_apps"
+
+    result = NavBar.items([nav_item])
+    assert_equal 1,  result.size
+    assert_equal "layouts/nav/all_apps",  result[0].partial_path
+  end
+
+  test "NavBar.items should return navigation link when nav_item is a string matching one app token" do
+    nav_item = "sys/bc_jupyter"
+
+    result = NavBar.items([nav_item])
+    assert_equal 1,  result.size
+    assert_equal "layouts/nav/link",  result[0].partial_path
+    assert_equal "Jupyter Notebook",  result[0].title
+    assert_equal "/batch_connect/sys/bc_jupyter/session_contexts/new",  result[0].url
+  end
+
+  test "NavBar.items should return navigation group when nav_item is a string matching multiple app tokens" do
+    nav_item = "sys/bc_*"
+
+    result = NavBar.items([nav_item])
+    assert_equal 1,  result.size
+    assert_equal "layouts/nav/group",  result[0].partial_path
+    assert_equal "Apps",  result[0].title
+    assert_equal 4,  result[0].apps.size
+  end
+
+  test "NavBar.items should return navigation group when nav_item is a string matching an sys application category" do
```

Comment:
Can you add a similar test for case sensitivity too? I.e., `    nav_item = "interactive apps"`.  I just want to be sure we preserve that moving forward.

Reply:
Will do

Reply:
Thanks, I went ahead and merged this as it's a small thing and I just wanted it off my plate today.